pkg/resources/server_test.go

```diff
@@ -395,12 +404,24 @@ var _ = Describe("resourceServer tests", func() {
 			}()
 
 			s := &devPluginListAndWatchServerMock{}
+			s.SetContext(context.Background())
 			err = rs.ListAndWatch(nil, s)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(s.devices).To(Equal(rs.devs))
 			Expect(len(s.devices)).To(Equal(100))
 			Expect(s.devices[5].Health).To(Equal(pluginapi.Unhealthy))
 		})
+		It("Stop ListAndWatch when stream is closed", func() {
+			rs := resourceServer{resourceName: "fake", socketName: "fake.sock"}
+			s := &devPluginListAndWatchServerMock{}
+			ctx, cnacel := context.WithCancel(context.Background())
```

Comment:
Thanks for adding the UT !

small nit: cnacel -> cancel 

otherwise im LGTM.